src/osinfo.rs

```diff
@@ -0,0 +1,109 @@
+/*
+ * This file is part of Edgehog.
+ *
+ * Copyright 2022 SECO Mind Srl
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use std::collections::HashMap;
+
+use astarte_sdk::types::AstarteType;
+
+use crate::error::DeviceManagerError;
+
+/// get structured data for `io.edgehog.devicemanager.OSInfo` interface
+pub fn get_os_info() -> Result<HashMap<String, AstarteType>, DeviceManagerError> {
+    let paths = ["/etc/os-release", "/usr/lib/os-release"];
+
+    let paths = paths.iter().filter(|f| std::path::Path::new(f).exists());
+
+    if let Some(path) = paths.into_iter().next() {
+        let os = std::fs::read_to_string(path)?;
+        return get_os_info_impl(&os);
+    }
+
+    Err(DeviceManagerError::FatalError(
+        "No os-release file found".to_owned(),
+    ))
+}
+
+pub fn get_os_info_impl(os: &str) -> Result<HashMap<String, AstarteType>, DeviceManagerError> {
```

Comment:
`parse_os_info` is a clearer name. Also we might change this function to private since we don't need to expose it.